packages/react-components/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -193,15 +212,16 @@ const useLabelStyles = makeStyles({
     marginLeft: tokens.spacingHorizontalM,
   },
 
-  // Use a (negative) margin to account for the difference between the indicator's height and the label's line height.
-  // This prevents the label from expanding the height of the Checkbox, but preserves line height if the label wraps.
+  // Use a (negative) margin to account for the difference between the indicator's height and the label's line height,
+  // as well as accounting for the introduced padding for handling the "empty space" around the label.
+  // This prevents the label from expanding the height of the checkbox, but preserves line height if the label wraps.
   medium: {
-    marginTop: `calc((${indicatorSizeMedium} - ${tokens.lineHeightBase300}) / 2)`,
-    marginBottom: `calc((${indicatorSizeMedium} - ${tokens.lineHeightBase300}) / 2)`,
+    marginTop: `calc(-1 * ${tokens.spacingVerticalS} + (${indicatorSizeMedium} - ${tokens.lineHeightBase300}) / 2)`,
+    marginBottom: `calc(-1 * ${tokens.spacingVerticalS} + (${indicatorSizeMedium} - ${tokens.lineHeightBase300}) / 2)`,
   },
   large: {
-    marginTop: `calc((${indicatorSizeLarge} - ${tokens.lineHeightBase300}) / 2)`,
-    marginBottom: `calc((${indicatorSizeLarge} - ${tokens.lineHeightBase300}) / 2)`,
+    marginTop: `calc(-1 * ${tokens.spacingVerticalS} + (${indicatorSizeLarge} - ${tokens.lineHeightBase300}) / 2)`,
+    marginBottom: `calc(-1 * ${tokens.spacingVerticalS} + (${indicatorSizeLarge} - ${tokens.lineHeightBase300}) / 2)`,
   },
```

Comment:
IMO it would be better to remove the padding from root and put it on the individual slots instead of adding padding to both the label AND the root, then adding a negative margin to overlap it.

Reply:
Agreed and done!